src/bio/smd/smd_pool.c

```diff
@@ -271,6 +271,9 @@ smd_pool_list(d_list_t *pool_list, int *pools)
 	td.td_count = 0;
 	D_INIT_LIST_HEAD(&td.td_list);
 
+	if (!smd_db_ready())
+		return 0; /* There is no NVMe, smd will not be initialized */
+
```

Comment:
I'm wondering why other SMD queries didn't trigger the assert, so I checked current code, seems all other SMD queries schedule a ULT on target 0 (first VOS xstream) and checks NVMe context there.

Should we apply the same way to pool list query? @sydneyvanda Any thoughts?